### PR TITLE
Sidecar compatiblity

### PR DESCRIPTION
**Pull request by ryanemerson**

### CODE REVIEW

**Review by rigazilla — APPROVED** [review]
LGTM.
I've added some readability comments. 
Everything started from this line:
`ispnContainer.Env = append(*env, corev1.EnvVar{`
that sound confusing to me.
rest of the comments are to uniform a little how `append` is used.

File: controllers/config_listener.go [hunk 1]
```diff
@@ -15,6 +15,8 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
 )
 
+const InfinispanListenerContainer = "infinispan-listener"
```
[the remark this answers is not in the document]
Comment (ryanemerson) [reply controllers/config_listener.go:18]: Luckily we've never released anything with the listener pod yet, so it's not too late :slightly_smiling_face:

**Comment by ryanemerson** [comment]
Thanks for the feedback @rigazilla, updated.

**Review by rigazilla — APPROVED** [review]

**Comment by rigazilla** [comment]
merged thanks @ryanemerson !

**Pull request merged**